sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala

```diff
@@ -506,6 +506,18 @@ class ExplainSuite extends ExplainSuiteHelper with DisableAdaptiveExecutionSuite
       checkKeywordsExistsInExplain(df2, keywords = "[key1=value1, KEY2=VALUE2]")
     }
   }
+
+  test("SPARK-35183: CombineConcats should call transformAllExpressions") {
```

Comment:
It's a bit weird to check the plan using explain result. If we want to have an end-to-end test for this optimization, let's check `sql(...).queryExecution.optimizedPlan` directly.

Normally we only write UT to test optimizer changes, so I'm fine to not have an end-to-end test here.

Reply:
Done. Reverted. 

(The explain test was copied and modified from another [one](https://github.com/apache/spark/blob/f208d808819dae38739fa9acda9510a4b9763dde/sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala#L151-L159) testing the original rule.